### DISCUSSION TITLE
Make changing the view in QDate not "steal" focus when it's explicitly prevented

### DISCUSSION THREAD

**Discussion by ddenev** [quasarframework/quasar#14841] [General - Components / Directives / etc] [answered]
I have a `QDate` that opens within a `QMenu` when clicking in an input field: https://codepen.io/ddenev/pen/XWYjQKG

My use-case is:
- I want the `QMenu` to be shown/hidden when the input field is focused/blurred and
- I want the input field to stay in focus while I navigate within the `QDate`
 
Therefore I have setup a `@mousedown.capture` event handler on the `QDate` component that simply does `event.preventDefault()` to prevent the blur on the input field whenever I click anywhere within the `QDate` element.
This works fine when clicking on the `<` and `>` buttons for changing the month or the year:
![image](https://user-images.githubusercontent.com/1393766/200333519-4db57b9d-82a6-45a1-b9d2-254de4171bda.png)
but fails when clicking on the month or the year buttons:
![image](https://user-images.githubusercontent.com/1393766/200333567-074f32dc-12b6-4a58-bee3-e4afdaa595b5.png)

I think the reason for this behaviour is this code in `QDate.js` (lines 710-712 in Quasar v2.10.1):
```js
watch(view, () => {
  blurTargetRef.value !== null && blurTargetRef.value.focus()
})
```

My questions:

1. Is there a particular reason (that I am missing) to force the focus on that part of the `QDate` element when changing the view?
1. Would it be possible to make `QDate` somehow respect the fact that I don't want it to get focused when clicked? Maybe introduce a `no-focus` prop?

**Comment by rstoenescu** [comment, marked as answer]
Fix will be available in Quasar v2.10.2 and v1.22.2
